platform/jvm/rules.bzl

```diff
@@ -307,12 +307,13 @@ _assemble_zip_to_jvm_platform = rule(
 )
 
 
-# TODO: upgrade all to JDK17
-def native_jdk16():
+def native_jdk17():
     return select({
-        "@vaticle_dependencies//util/platform:is_mac": "@jdk16_mac//file",
-        "@vaticle_dependencies//util/platform:is_linux": "@jdk16_linux//file",
-        "@vaticle_dependencies//util/platform:is_windows": "@jdk16_windows//file",
+        "@vaticle_bazel_distribution//platform:is_linux_arm64": "@jdk17_linux_arm64//file",
+        "@vaticle_bazel_distribution//platform:is_linux_x86_64": "@jdk17_linux_x86_64//file",
+        "@vaticle_bazel_distribution//platform:is_mac_arm64": "@jdk17_mac_arm64//file",
+        "@vaticle_bazel_distribution//platform:is_mac_x86_64": "@jdk17_mac_x86_64//file",
+        "@vaticle_bazel_distribution//platform:is_windows_x86_64": "@jdk17_windows_x86_64//file",
```

Comment:
This is technically still a cycle — `@jdk17_*` are defined in dependencies, but they can come from anywhere. We are depending on a specific naming convention though.